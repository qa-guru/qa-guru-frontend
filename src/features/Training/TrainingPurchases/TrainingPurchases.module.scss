.link {
  text-decoration: none;
}

.paper {
  padding: 24px 24px 15px 24px;
  min-height: 116px;
}
