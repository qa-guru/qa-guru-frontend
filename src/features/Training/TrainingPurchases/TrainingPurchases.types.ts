import { TrainingPurchasesQuery } from "../../../generated/graphql";

export interface ITrainings {
  data: TrainingPurchasesQuery;
}
