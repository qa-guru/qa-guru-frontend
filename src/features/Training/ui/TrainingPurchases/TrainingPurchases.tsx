import React from "react";
import { Link } from "react-router-dom";
import { CardActionArea, Grid, Paper, Typography } from "@mui/material";
import { TrainingPurchasesQuery } from "../../../../generated/graphql";
import styles from "./TrainingPurchases.module.scss";

interface ITrainings {
  data: TrainingPurchasesQuery;
}

const TrainingPurchases: React.FC<ITrainings> = ({ data }) => {
  const { trainingPurchases } = data;

  return (
    <Grid container spacing={2} columns={{ xs: 6, md: 12 }}>
      {trainingPurchases?.map((item, index) => {
        return (
          <Grid item xs={6} key={index}>
            <Link
              className={styles.link}
              to={`/training/${item?.trainingTariff?.training?.id}`}
            >
              <CardActionArea>
                <Paper className={styles.paper}>
                  <Typography variant="h3">
                    {item?.trainingTariff?.training?.name}
                  </Typography>
                  <Typography
                    color="primary"
                    variant="h4"
                    mt="20px"
                    align="right"
                  >
                    Продолжить
                  </Typography>
                </Paper>
              </CardActionArea>
            </Link>
          </Grid>
        );
      })}
    </Grid>
  );
};

export default TrainingPurchases;
