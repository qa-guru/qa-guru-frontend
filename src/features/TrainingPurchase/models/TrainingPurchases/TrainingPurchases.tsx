import React from "react";
import { Link } from "react-router-dom";
import { Box, CardActionArea, Paper, Typography } from "@mui/material";
import styles from "./TrainingPurchases.module.scss";
import { TrainingPurchasesQuery } from "../../../../generated/graphql";

interface ITrainingByUserId {
  data: TrainingPurchasesQuery;
}

const TrainingPurchases: React.FC<ITrainingByUserId> = ({ data }) => {
  return (
    <Box className={styles.box}>
      {data?.trainingPurchases?.map((item, index) => {
        return (
          <Link
            className={styles.link}
            to={`/training/${item?.trainingTariff?.training?.id}`}
            key={index}
          >
            <CardActionArea>
              <Paper className={styles.paper}>
                <Typography className={styles.title}>
                  {item?.trainingTariff.name}
                </Typography>
                <Typography className={styles.description}>
                  {item?.trainingTariff.description}
                </Typography>
              </Paper>
            </CardActionArea>
          </Link>
        );
      })}
    </Box>
  );
};

export default TrainingPurchases;
