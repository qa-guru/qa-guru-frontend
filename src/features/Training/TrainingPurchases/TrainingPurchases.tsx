import React from "react";
import { Link } from "react-router-dom";
import { CardActionArea, Grid, Paper, Typography } from "@mui/material";
import { ITrainings } from "./TrainingPurchases.types";

const TrainingPurchases: React.FC<ITrainings> = ({ data }) => {
  const { trainingPurchases } = data;

  return (
    <Grid container spacing={2} columns={{ xs: 6, md: 12 }}>
      {trainingPurchases?.map((item, index) => {
        return (
          <Grid item xs={6} key={index}>
            <Link
              style={{ textDecoration: "none" }}
              to={`/training/${item?.trainingTariff?.training?.id}`}
            >
              <CardActionArea>
                <Paper sx={{ padding: "24px 24px 15px 24px" }}>
                  <Typography variant="h3">
                    {item?.trainingTariff?.training?.name}
                  </Typography>
                  <Typography
                    color="primary"
                    variant="h4"
                    mt="20px"
                    align="right"
                  >
                    Продолжить
                  </Typography>
                </Paper>
              </CardActionArea>
            </Link>
          </Grid>
        );
      })}
    </Grid>
  );
};

export default TrainingPurchases;
