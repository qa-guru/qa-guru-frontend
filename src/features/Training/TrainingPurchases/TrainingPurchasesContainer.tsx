import React from "react";
import Spinner from "../../../shared/Spinner";
import { useTrainingPurchasesQuery } from "../../../api/graphql/trainingPurchase/trainingPurchases";
import TrainingPurchases from "./TrainingPurchases";

const TrainingPurchasesContainer: React.FC = () => {
  const { data, loading } = useTrainingPurchasesQuery();

  if (loading && !data) return <Spinner />;

  return <TrainingPurchases data={data!} />;
};

export default TrainingPurchasesContainer;
