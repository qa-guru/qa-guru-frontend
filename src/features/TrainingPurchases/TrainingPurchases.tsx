import React from "react";
import { Link } from "react-router-dom";
import { CardActionArea, Grid, Paper, Typography } from "@mui/material";
import { ITrainings } from "./TrainingPurchases.types";

const style = {
  paper: { padding: { xs: "15px", md: "20px" } },
};

const TrainingPurchases: React.FC<ITrainings> = ({ data }) => {
  const { trainingPurchases } = data;

  return (
    <>
      <Typography variant="h4" mb="20px">
        Мои курсы
      </Typography>
      <Grid container spacing={2} columns={{ xs: 6, md: 12 }}>
        {trainingPurchases?.map((item, index) => {
          const { id, name } = item!.trainingTariff.training!;

          return (
            <Grid item xs={6} key={index}>
              <Link style={{ textDecoration: "none" }} to={`/training/${id}`}>
                <CardActionArea>
                  <Paper sx={style.paper}>
                    <Typography variant="h5">{name}</Typography>
                    <Typography
                      color="primary"
                      variant="subtitle2"
                      mt="20px"
                      align="right"
                    >
                      Продолжить
                    </Typography>
                  </Paper>
                </CardActionArea>
              </Link>
            </Grid>
          );
        })}
      </Grid>
    </>
  );
};

export default TrainingPurchases;
