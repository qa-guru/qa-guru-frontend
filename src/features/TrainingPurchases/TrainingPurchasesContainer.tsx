import React from "react";
import TrainingPurchases from "./TrainingPurchases";
import Spinner from "../../shared/Spinner/Spinner";
import { useTrainingPurchasesQuery } from "../../api/graphql/trainingPurchase/trainingPurchases";
import NoDataErrorMessage from "../../shared/NoDataErrorMessage";

const TrainingPurchasesContainer: React.FC = () => {
  const { data, loading } = useTrainingPurchasesQuery();

  if (loading) return <Spinner />;
  if (!data) return <NoDataErrorMessage />;

  return <TrainingPurchases data={data} />;
};

export default TrainingPurchasesContainer;
