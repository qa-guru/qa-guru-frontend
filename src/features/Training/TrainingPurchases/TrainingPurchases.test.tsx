import { MemoryRouter } from "react-router-dom";
import { render } from "../../../test/utilsTest";
import TrainingPurchases from "./TrainingPurchases";
import { TrainingPurchasesQuery } from "../../../generated/graphql";

const mockData: TrainingPurchasesQuery = {
  __typename: "Query",
  trainingPurchases: [
    {
      __typename: "TrainingPurchaseDto",
      id: "string",
      user: {
        __typename: "UserDto",
        id: "string",
        email: "string",
        firstName: "string",
        lastName: "string",
        middleName: "string",
      },
      trainingTariff: {
        __typename: "TrainingTariffDto",
        id: "string",
        name: "string",
        code: "string",
        price: 7,
        homeWork: false,
        description: "string",
        training: {
          __typename: "TrainingDto",
          id: "string",
          name: "string",
        },
      },
    },
  ],
};

describe("TrainingPurchases", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <TrainingPurchases data={mockData!} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
