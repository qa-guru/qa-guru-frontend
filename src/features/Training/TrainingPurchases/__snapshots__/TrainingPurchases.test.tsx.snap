// Vitest Snapshot v1

exports[`TrainingPurchases > the component is render 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1tp3ba8-MuiGrid-root"
    >
      <a
        href="/training/string"
        style="text-decoration: none;"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-6zsig3-MuiPaper-root"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              string
            </h3>
            <h4
              class="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight css-6b8kcz-MuiTypography-root"
            >
              Продолжить
            </h4>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
